### DISCUSSION TITLE
Launching composable nodes

### DISCUSSION THREAD

**Discussion by thomasegriffith** [mvukov/rules_ros2#121] [General]
Has anybody had success with launching composable nodes?

I'm following [this python example](https://docs.ros.org/en/humble/How-To-Guides/Launching-composable-nodes.html) with a launch target of: 

```
ros2_launch(
    name = "composable_example_launch",
    launch_file = "composable_example_launch.py",
    nodes = ["//path/to/some/random:node"],
)
```

I run into the below error when `bazel run`ning the above launch file:

```
[ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'rclcpp_components' not found, searching: ['path/to/launch/file/composable_example_launch_launch_ament_setup']"
```

Any ideas how to get around this?

TIA!

**Comment by mvukov** [comment]
A rclcpp component basically behaves like a plugin. It looks like it needs a kind of a special registration. At the moment this doesn't work with rules_ros2.

**Comment by thomasegriffith** [comment]
Thanks!  Is there anything I can help with, to get support added for this?

**Comment by mvukov** [comment]
Sure, it needs to be figured out how to register a component within the ament file-tree.

**Comment by thomasegriffith** [comment]
Cool.  No guarantees, but I'll try to find some weekend to work on it.
